packages/amplify-cli/src/domain/amplify-telemetry/Telemetry.ts

```diff
@@ -0,0 +1,83 @@
+import uuid from 'uuid';
+import { Input } from '../input';
+import https from 'https';
+import { UrlWithStringQuery } from 'url';
+import redactInput from './identifiable-input-regex';
+import { TelemetryPayload } from './TelemetryPayload';
+import { getUrl } from './getTelemetryUrl';
+import { ITelemetry } from './ITelemetry';
+
+export class Telemetry implements ITelemetry {
+  sessionUuid: String;
+  installationUuid: String = '';
+  version: String = '';
+  input: Input;
+  url: UrlWithStringQuery;
+  private static instance: Telemetry;
+
+  private constructor() {
+    this.sessionUuid = uuid.v4();
+    this.url = getUrl();
+    this.input = new Input([]);
+  }
+
+  init(installationUuid: String, version: String, input: Input): void {
+    this.installationUuid = installationUuid;
+    this.version = version;
+    this.input = redactInput(input, true);
+  }
+
+  static get Instance(): ITelemetry {
```

Comment:
Why do we need static instance variable in the `ITelemetry` implementations?

This only accessed through `context`, which already has an instance of those classes.

Reply:
This is how a singleton is created